examples/jsm/nodes/ShaderNode.js

```diff
@@ -208,66 +211,83 @@ export const label = ( node, name ) => {
 
 export const temp = ( node ) => nodeObject( new VarNode( nodeObject( node ) ) );
 
-const ConvertType = function ( type, valueClass = null, valueComponents = 1 ) {
+const cacheMap = new Map();
 
-	return ( ...params ) => {
+const ConvertType = function ( type ) {
 
-		if ( params[ 0 ]?.isNode === true ) {
+	return ( ...params ) => {
 
-			return nodeObject( new ConvertNode( params[ 0 ], type ) );
+		let node;
 
-		}
+		if ( params.length === 0 ) {
 
-		if ( ( params.length === 1 ) && ( valueComponents !== 1 ) ) {
+			node = nodeObject( new ConstNode( getValueFromType( type ), type ) );
 
-			// Providing one scalar value: This value is used for all components
+		} else {
 
-			for ( let i = 1; i < valueComponents; i ++ ) {
+			if ( type === 'color' && params.length === 1 ) {
```

Comment:
See https://github.com/mrdoob/three.js/issues/23634.

Reply:
So should it actually be `params.length === 3` here?

Reply:
the use of `color()` should be `color( 0xff00ff )`

Reply:
Then I think it should be [this](https://github.com/mrdoob/three.js/pull/23635/commits/1c0494f5b5af1bac5e876136fa5526b05a9f965a).